deployment/test/json-test.js

```diff
@@ -0,0 +1,332 @@
+      deployment
+    );
+  }
+
+  // Make sure there is at least one deployment ID
+  if (Object.keys(deploymentData["deployment-ids"]).length == 0) {
+    throw (
+      "Please specify at least one deployment-id in deployment.json for protocol: " +
+      protocol +
+      " deployment: " +
+      deployment
+    );
+  }
+
+  // Make sure all deployment IDs are valid
+  for (const [deploymentID, location] of Object.entries(
+    deploymentData["deployment-ids"]
+  )) {
+    if (
+      !["hosted-service", "cronos-portal", "decentralized-network"].includes(
+        deploymentID
+      )
+    ) {
+      throw (
+        "Invalid deployment-id in deployment.json for protocol: " +
+        protocol +
+        " deployment: " +
+        deployment +
+        " deployment-id: " +
+        deploymentID
+      );
+    }
+
+    if (!location) {
+      throw (
+        "Please specify deployment-id in deployment.json for protocol: " +
+        protocol +
+        " deployment: " +
+        deployment +
+        " deployment-id: " +
+        deploymentID
+      );
+    }
+  }
+}
+
+// Makes sure the version specified in the json file is valid.
+function checkValidVersion(version) {
+  if (!version) {
+    throw "See deployment.json: (0) version is missing";
+  }
+
+  // Make sure version lenght is 3 (major.minor.patch)
+  if (version.split(".").length != 3) {
+    throw "See deployment.json: (1) version is not valid - should be 3 integers separated by periods - must be in format x.x.x (e.g. 1.3.1)";
+  }
+
+  // Make sure each integer is valid
+  let array = version.split(".");
+  if (
+    !array.every((element) => {
+      return !isNaN(element);
+    })
+  ) {
+    throw "See deployment.json: (2) version is not valid - make sure to use integers between the periods - must be in format x.x.x (e.g. 1.3.1)";
+  }
+}
+
+function checkTemplatePresent(protocol, deployment, deploymentData) {
+  if (!deploymentData["files"]) {
+    throw (
+      "Please specify files in deployment.json for protocol: " +
+      protocol +
+      " deployment: " +
+      deployment
+    );
+  }
+
+  // Also check that the template is present since it is required for subgraph prep.
+  if (!deploymentData["files"]["template"]) {
+    throw (
+      "Please specify template in deployment.json for protocol: " +
+      protocol +
+      " deployment: " +
+      deployment
+    );
+  }
+}
+
+function checkOptionsPresent(protocol, deployment, deploymentData) {
+  if (!deploymentData["options"]) {
+    throw (
+      "Please specify options in deployment.json for protocol: " +
+      protocol +
+      " deployment: " +
+      deployment
+    );
+  }
+}
+
+function checkDuplicateIDs(deploymentJsonMap) {
+  let deploymentIDs = new Set();
+  let protocols = new Set();
+
+  // Check for duplicate deployment IDs
+  for (const [protocol, protocolData] of Object.entries(deploymentJsonMap)) {
+    protocols.add(protocol);
+    for (const [deployment, deploymentData] of Object.entries(
```

Comment:
I left this test in, since there is still some utility for duplicates across protocols. I will be changing it as we discussed to use a validation tool in the near future.